lib/src/model.dart

```diff
@@ -3554,87 +3554,110 @@ abstract class ModelElement extends Canonicalization
     final RegExp basicAnimationRegExp =
         new RegExp(r'''{@animation\s+([^}]+)}''');
 
-    // Animations have four parameters, and the last one can be surrounded by
-    // quotes (which are ignored). This RegExp is used to validate the directive
-    // for the correct number of parameters.
-    final RegExp animationRegExp =
-        new RegExp(r'''{@animation\s+([^}\s]+)\s+([^}\s]+)\s+([^}\s]+)'''
-            r'''\s+['"]?([^}]+)['"]?}''');
-
     // Matches valid javascript identifiers.
-    final RegExp validNameRegExp = new RegExp(r'^[a-zA-Z_][a-zA-Z0-9_]*$');
-
-    // Keeps names unique.
-    final Set<String> uniqueNames = new Set<String>();
+    final RegExp validIdRegExp = new RegExp(r'^[a-zA-Z_]\w*$');
+
+    final Set<String> uniqueIds = new Set<String>();
+    String getUniqueId(String base) {
+      int count = 1;
+      String id = '$base$count';
+      while (uniqueIds.contains(id)) {
+        count++;
+        id = '$base$count';
+      }
+      return id;
+    }
 
     return rawDocs.replaceAllMapped(basicAnimationRegExp, (basicMatch) {
-      final Match match = animationRegExp.firstMatch(basicMatch[0]);
-      if (match == null) {
+      final ArgParser parser = new ArgParser();
+      parser.addOption('id');
+      final ArgResults args = _parseArgs(basicMatch[1], parser);
+      final List<String> positionalArgs = args.rest.sublist(0);
+      String uniqueId;
+      bool wasDeprecated = false;
+      if (positionalArgs.length == 4) {
+        // Supports the original form of the animation tag for backward
+        // compatibility.
+        uniqueId = positionalArgs.removeAt(0);
```

Comment:
This will of course completely discard the name.  Given that anchors do not actually work yet I suppose this is probably harmless?

Reply:
Right, got it.